### PR TITLE
Improve write_to_waveform_memory

### PR DESCRIPTION
**Pull request by ghost**
Description:
Add new improvements to write_to_waveform_memory:
* Append to existing transactions
* Add a flag to disable local santity check

Add small improvement to read_from_waveform_memory.

Checklist:

- [x] Add tests for the change to show correct behavior.
- [x] Add or update relevant docs, code and examples.
- [x] Update CHANGELOG.rst with relevant information and add the issue number.
- [x] Add .. versionchanged:: where necessary.

### CODE REVIEW

File: tests/test_awg.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import json
 from collections import OrderedDict
 from itertools import cycle
+from tabnanny import check
```
Comment (markheik) [inline tests/test_awg.py:4]: Unnecessaru import?

File: src/zhinst/toolkit/driver/nodes/awg.py [hunk 2]
```diff
@@ -167,47 +170,60 @@ def load_sequencer_program(
...
         """Writes waveforms to the waveform memory.
 
         The waveforms must already be assigned in the sequencer program.
 
         Args:
             waveforms: Waveforms that should be uploaded.
+            indexes: Specify a list of indexes that should be uploaded. If
+                nothing is specified all available indexes in waveforms will
+                be uploaded. (default = None)
+            check: Enable sanity check preformed by toolkit, based on the
```
Comment (markheik) [inline src/zhinst/toolkit/driver/nodes/awg.py:184]: Would `validate` be better argument name? As Toolkit validates the  waveforms.

Comment (tobiasah) [reply src/zhinst/toolkit/driver/nodes/awg.py]: good point

File: src/zhinst/toolkit/driver/nodes/awg.py [hunk 3]
```diff
@@ -167,47 +170,60 @@ def load_sequencer_program(
...
+        num_waveforms = None
+        if check:
+            waveform_info = json.loads(self.waveform.descriptors()).get("waveforms", [])
+            num_waveforms = len(waveform_info)
+        with create_or_append_set_transaction(self._root):
+            for waveform_index in waveforms.keys():
+                if indexes and waveform_index not in indexes:
+                    continue
+                if num_waveforms is not None and waveform_index >= num_waveforms:
+                    raise IndexError(
```
[the remark this answers is not in the document]
Comment (tobiasah) [reply src/zhinst/toolkit/driver/nodes/awg.py:205]: I would change this once we takle the exceptions

[the remark this answers is not in the document]
Comment (markheik) [reply src/zhinst/toolkit/driver/nodes/awg.py:205]: Can you add an TODO to this error? As changing it later would then break the API.

**Review by markheik — APPROVED** [review]

**Pull request merged**